repl/src/main/scala/com/cloudera/livy/repl/ReplDriver.scala

```diff
@@ -83,4 +84,29 @@ class ReplDriver(conf: SparkConf, livyConf: RSCConf)
     return session.state.toString
   }
 
+  override protected def createWrapper(msg: BaseProtocol.BypassJobRequest): BypassJobWrapper = {
+    interpreter match {
+      case pi: PythonInterpreter => pi.createWrapper(this, msg)
+      case _ => super.createWrapper(msg)
+    }
+  }
+
+  override protected def addFile(path: String): Unit = {
+    interpreter match {
+      case pi: PythonInterpreter => pi.addFile(path)
+      case _ => super.addFile(path)
+    }
+  }
+
+  override protected def addJarOrPyFile(path: String): Unit = {
+    path.endsWith(".jar") match {
```

Comment:
In these cases it's better to just use `if...else`.

Is it common to distribute python code in files ending with ".jar"? pyspark lists '.jar' in the list of known package extensions that are loaded in the python interpreter. So maybe in that case you should call both `super.addJarOrPyFile` and `pi.addPyFile`?